src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -322,7 +322,8 @@ def _build_storage_profile():
                     'createOption': 'fromImage',
                     'name': os_disk_name,
                     'caching': os_caching,
-                    'managedDisk': {'storageAccountType': disk_info['os'].get('storageAccountType')}
+                    'managedDisk': {'storageAccountType': disk_info['os'].get('storageAccountType')},
+                    'diffDiskSettings': disk_info['os']['diffDiskSettings']
```

Comment:
I suggest you only add this entry if we do have the configuration. This will prevent the none entry from breaking Azure Stack